### PR TITLE
Create verbose client debugger

### PR DESCRIPTION
**Pull request by alexcb**
when running under debug mode, log all interactions with the buildkit
gateway client.

Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

File: examples/tests/Earthfile [hunk 1]
```diff
@@ -582,8 +582,10 @@ cache-mount-arg:
     RUN cat output.txt | grep '\*cached\* --> RUN echo Doing something 2'; test "$?" != 0
 
 true-false-flag:
+    RUN --no-cache rm -f /var/log/buildkitd.log output.txt
```
Comment (vladaionescu) [inline examples/tests/Earthfile:585]: What's this for?

**Review by vladaionescu — APPROVED** [review]
Would be interesting to see some sample output if you have it handy

File: examples/tests/Earthfile (hunk 1 above)
Comment (alexcb) [reply examples/tests/Earthfile:585]: oh no, clearly wasn't supposed to have been added :facepalm:

**Review by vladaionescu — APPROVED** [review]

**Comment by alexcb** [comment]
> Would be interesting to see some sample output if you have it handy

Here's an example of one call to Solve; it's definitely verbose.

```
Solve req={
	"Evaluate": false,
	"Definition": {
		"def": [
			"GmcKK2RvY2tlci1pbWFnZTovL2RvY2tlci5pby9hbHBpbmUvZ2l0OnYyLjMwLjESHAoQaW1hZ2UucmVjb3JkdHlwZRIIaW50ZXJuYWwSGgoRaW1hZ2UucmVzb2x2ZW1vZGUSBWxvY2FsUg4KBWFtZDY0EgVsaW51eFoA",
			"GqoFCk5naXQ6Ly9naXRodWIuY29tL2VhcnRobHkvYnVpbGRraXQuZ2l0IzE5OWFkNmE1YzIxM2QyYTY5MzdjZWQ5ZTJiNTJiNWI1N2UwYTNhMzcSJwoUZ2l0LmF1dGhoZWFkZXJzZWNyZXQSD0dJVF9BVVRIX0hFQURFUhIlChNnaXQuYXV0aHRva2Vuc2VjcmV0Eg5HSVRfQVVUSF9UT0tFThIyCgtnaXQuZnVsbHVybBIjZ2l0QGdpdGh1Yi5jb206ZWFydGhseS9idWlsZGtpdC5naXQSFgoOZ2l0LmtlZXBnaXRkaXISBHRydWUSngMKEWdpdC5rbm93bnNzaGhvc3RzEogDZ2l0aHViLmNvbSBzc2gtcnNhIEFBQUFCM056YUMxeWMyRUFBQUFCSXdBQUFRRUFxMkE3aFJHbWRubTl0VURiTzlJRFN3Qks2VGJRYStQWFlQQ1B5NnJiVHJUdHc3UEhrY2NLcnBwMHlWaHA1SGRFSWNLcjZwTGxWREJmT0xYOVFVc3lDT1Ywd3pmaklKTmxHRVlzZGxMSml6SGhibjJtVWp2U0FIUXFaRVRZUDgxZUZ6TFFOblBIdDRFVlZVaDdWZkRFU1U4NEtlem1ENVFsV3BYTG12VTMxL3lNZitTZTh4aEhUdktTQ1pJRkltV3dvRzZtYlVvV2Y5bnpwSW9hU2pCK3dlcXFVVW1wYWFhc1hWYWw3MkorVVgyQisyUlBXM1JjVDBlT3pRZ3FsSkwzUktyVEp2ZHNqRTNKRUF2R3EzbEdIU1pYeTI4RzNza3VhMlNtVmkvdzR5Q0U2Z2JPRHFuVFdsZzcrd0M2MDR5ZEdYQThWSmlTNWFwNDNKWGlVRkZBYVE9PQoSGwoQZ2l0Lm1vdW50c3Noc29jaxIHZGVmYXVsdFoA",
			"CkkKR3NoYTI1Njo4NDQ1MGUyZDY4YTliMGM2Yjk5NDc2ZTVjZmFjMmE4ODcxZGU1MWNlMDY2MjhlNzhlNWRhZTYxZDI4ODRmYjhlCkkKR3NoYTI1NjplMzA1MzBmZmQyNTAzZTU4NTMzM2YzMDAwODU1MDljMWMwMWZlOGUzZjg2Nzg0ZDU2ZThmMWJjYzRhNTJhNjUxEpsCCtcBCgcvYmluL3NoCgItYwq9AWdpdCByZXYtcGFyc2UgSEVBRCA+L2Rlc3QvZ2l0LWhhc2ggOyBnaXQgcmV2LXBhcnNlIC0tYWJicmV2LXJlZiBIRUFEID4vZGVzdC9naXQtYnJhbmNoICB8fCB0b3VjaCAvZGVzdC9naXQtYnJhbmNoIDsgZ2l0IGRlc2NyaWJlIC0tZXhhY3QtbWF0Y2ggLS10YWdzID4vZGVzdC9naXQtdGFncyB8fCB0b3VjaCAvZGVzdC9naXQtdGFncxoIL2dpdC1zcmMSEBoBLyD///////////8BKAESEgj///////////8BGgUvZGVzdBIZCAEaCC9naXQtc3JjIP///////////wEoAVIOCgVhbWQ2NBIFbGludXhaAA==",
			"CkkKR3NoYTI1Njo5ODFiOWNhNjUxYTAxOWFlNWQ1MmVlNjJjMWQyYmFmMDliMWJiYWU3MjJhYWRhOWNjMWYzMjc4NDdlNDdmNjA0"
		],
		"metadata": {
			"sha256:84450e2d68a9b0c6b99476e5cfac2a8871de51ce06628e78e5dae61d2884fb8e": {
				"caps": {
					"source.image": true
				}
			},
			"sha256:981b9ca651a019ae5d52ee62c1d2baf09b1bbae722aada9cc1f327847e47f604": {
				"description": {
					"llb.customname": "[internal] GET GIT META github.com/earthly/buildkit:199ad6a5c213d2a6937ced9e2b52b5b57e0a3a37"
				},
				"caps": {
					"exec.meta.base": true,
					"exec.mount.bind": true
				}
			},
			"sha256:e30530ffd2503e585333f300085509c1c01fe8e3f86784d56e8f1bcc4a52a651": {
				"description": {
					"llb.customname": "[internal] GIT CLONE git@github.com:earthly/buildkit.git"
				},
				"caps": {
					"source.git": true,
					"source.git.fullurl": true,
					"source.git.keepgitdir": true,
					"source.git.knownsshhosts": true,
					"source.git.mountsshsock": true
				}
			},
			"sha256:e935bc34a69581678afd182083163abd34bc392284f933010475bebf5f80dbde": {
				"caps": {
					"constraints": true,
					"meta.description": true,
					"platform": true
				}
			}
		},
		"Source": {
			"locations": {
				"sha256:84450e2d68a9b0c6b99476e5cfac2a8871de51ce06628e78e5dae61d2884fb8e": {},
				"sha256:981b9ca651a019ae5d52ee62c1d2baf09b1bbae722aada9cc1f327847e47f604": {},
				"sha256:e30530ffd2503e585333f300085509c1c01fe8e3f86784d56e8f1bcc4a52a651": {}
			}
		}
	},
	"Frontend": "",
	"FrontendOpt": null,
	"FrontendInputs": null,
	"CacheImports": null
} res={
	"Ref": {},
	"Refs": null,
	"Metadata": null
}; err=<nil>
```

If it's too verbose, another possibility to add a `--trace` flag?

**Pull request merged**